### DISCUSSION TITLE
About the tutorial pitzDaily

### DISCUSSION THREAD

**Discussion by terumasakitayama** [mdolab/dafoam#550] [General]
Hello.

I am trying topology optimization with the tutorial pitzDaily, what is the best way to impose volume constraints?

I have defined leList and teList and am calling addVolumeConstraint on pyGeometry but it is not working.

**Comment by friedenhe** [comment]
Hmm. the pitzDaily is a topology optimization, so the volume does not change during the optimization.. Can you elaborate on what volume constraint you plan to impose?

**Reply by terumasakitayama** [reply]
Sorry, I misunderstood.

I have one more question.
I think the alphaPorosity of pitzDaily is calculated as follows.

![image](https://github.com/mdolab/dafoam/assets/70757835/8ff714e5-0b56-49fd-92ad-61eae7ae660a)


I would like to display the distribution of gamma instead of alpha.

**Comment by friedenhe** [comment]
No, we don't have the above equation implemented in DAFoam.. We just added alpha directly to the momentum equation.
